### DISCUSSION TITLE
How can I access script files e.g. paperless via a shell on a Proxmox-Server

### DISCUSSION THREAD

**Discussion by bergquelle** [tteck/Proxmox#3027] [Help Needed] [answered]
Hello!

I installed Proxmox on a standalone Notebook and I used the script from a github user TTECK (https://github.com/tteck/Proxmox) 
for installing Home Assistant and paperless-ngx. Everything works fine so far.

Now I need to modify certain script files and I do not know how to access them via proxmox. 
Using a shell in Proxmox does not help me at all, because I cannot find the place where e.g. paperless-ngx is located. 
BTW - I am not an expert in UNIX.

Hopefully I can find some help here.

Thanks a lot.
Volker

**Comment by Mellowlynx** [comment, marked as answer]
If you open the Console of the paperless-ngx container, you can use nano to edit files.
For example, the main config file of paperless-ngx is located in /opt/paperless.
So use: `nano /opt/paperless/paperless.conf`

When done;
Ctrl + R to Write/Save
Ctrl + X to Exit 

You probably need to restart to load the updated config, to do this type: `reboot`

![afbeelding](https://github.com/tteck/Proxmox/assets/2348339/792acd61-de26-49c1-993a-7a20dc3770ca)

For Home Assistant, I would like to recommend the "File editor" add-on.
https://www.home-assistant.io/getting-started/configuration/

Hope this helps.

**Comment by bergquelle** [comment]
Hi Mellowlynx, thanks for your quick reply.

I tried it and it works fine.

Thanks

**Reply by Mellowlynx** [reply]
You're welcome 😃

**Reply by bergquelle** [reply]
Hello Mellowlynx, 
can I ask you directly some questions or should I ask the community?
As you know I run Proxmox with paperless and Home Assitant, which works fine.
I also have a Synology NAS and I would like to backup e.g. paperless, in particular, the original docs.
What would be the easiest way to do that? 
Maybe a script could connect to the NAS and copy certain files to it - does that make sence or are there better ways.
Thanks for your help.

**Reply by Mellowlynx** [reply]
I use the built-in Proxmox backup system to backups all my VM's and LXC's to my NAS.
To be able to do this, I mounted the share in the cluster (under Storage), so all nodes have access to it.
![afbeelding](https://github.com/tteck/Proxmox/assets/2348339/70521010-4843-41d2-914d-c2f21cae0cb3)

Then under Backup, set up a schedule that you like and set storage to the mounted folder.
![afbeelding](https://github.com/tteck/Proxmox/assets/2348339/004ee124-5951-4275-8ec5-572b77158cd7)


I run 3 types of backups.
- One, twice a week, that backups almost everything.
- One, that backup two VM's only once a week.
- Before most config changes and all testing, upgrades, etc. I make a manual extra backup. (the note will include why I made it)

Because the backup folder is mounted on every node, I can restore an LXC in seconds on another node (with two exceptions, the LXC's that have a second drive.)


You could also add the NAS share as a resource to the LXC and copy/rsync data to/from it from within the LXC, using a cronjob or something.

I hope this helps :)

**Reply by bergquelle** [reply]
Hello, thanks for your feedback. 

I will play around with what you mentioned - it seems to be a failry good and quick.

:-)
